tests/integ/modin/index/test_index_methods.py

```diff
@@ -212,3 +219,179 @@ def test_df_index_shape(native_df):
+        assert_series_equal(
+            native_index.to_series(
+                index=range(10, 10 + len(native_index)), name="name"
+            ),
+            snow_index.to_series(index=range(10, 10 + len(snow_index)), name="name"),
+            check_index_type=False,
+        )
+
+
+@sql_count_checker(query_count=4, join_count=1)
+@pytest.mark.parametrize("native_df", TEST_DFS)
+def test_df_index_columns_to_series(native_df):
+    snow_df = pd.DataFrame(native_df)
+
+    # Snowpark pandas sets the dtype of df.columns for an empty df as object and native pandas sets it as int,
+    # so we avoid this check on an empty df
+    if native_df.empty:
+        check_dtype = False
+    else:
+        check_dtype = True
+
+    assert_series_equal(
+        snow_df.index.to_series(),
+        native_df.index.to_series(),
+        check_index_type=False,
+    )
+    assert_series_equal(
+        snow_df.columns.to_series(),
+        native_df.columns.to_series(),
+        check_dtype=check_dtype,
+        check_index_type=False,
+    )
+
+    with SqlCounter(join_count=1):
+        assert_series_equal(
+            snow_df.index.to_series(index=range(len(native_df.index)), name=1),
+            native_df.index.to_series(index=range(len(native_df.index)), name=1),
+            check_index_type=False,
+        )
+
+    assert_series_equal(
+        snow_df.columns.to_series(index=range(len(native_df.columns)), name=True),
+        native_df.columns.to_series(index=range(len(native_df.columns)), name=True),
+        check_dtype=check_dtype,
+        check_index_type=False,
+    )
+
+
+@sql_count_checker(query_count=2)
+@pytest.mark.parametrize("native_index", NATIVE_INDEX_TEST_DATA)
+def test_index_to_frame(native_index):
+    snow_index = pd.Index(native_index)
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+        snow_index.to_frame(),
+        native_index.to_frame(),
+        check_index_type=False,
+        check_column_type=False,
+    )
+
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+        snow_index.to_frame(index=False, name="name"),
+        native_index.to_frame(index=False, name="name"),
+        check_index_type=False,
+        check_column_type=False,
+    )
+
+
+@sql_count_checker(query_count=4)
+@pytest.mark.parametrize("native_df", TEST_DFS)
+def test_df_index_columns_to_frame(native_df):
+    snow_df = pd.DataFrame(native_df)
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+        snow_df.index.to_frame(),
+        native_df.index.to_frame(),
+        check_index_type=False,
+        check_column_type=False,
+    )
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+        snow_df.columns.to_frame(),
+        native_df.columns.to_frame(),
+        check_index_type=False,
+        check_column_type=False,
+    )
+
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+        snow_df.index.to_frame(index=False, name=1),
```

Comment:
can you put index value, and name value as parameters, similar as how you put native df? also, do we need to test over all dfferent dfs, or shall we test with different NATIVE_INDEX_TEST_DATA